### ISSUE TITLE
Fetching folder error on android 11 Oneplus Nord 2

### ISSUE THREAD

**Issue by chetananakum31** [Mindinventory/Lassi-Android#46]
I have found that there is issue on `folderViewModel.fetchFolders()` function. I am using lates version of the library.

I have tested on OnePlus Nord 2 (having Android 11 OS) app is crashed.
Same I have tested on OnePlust 7 (having Android 11 OS) but app works fine.

Please let me know if you need more details.

**Comment by Nkgohil007** [comment]
Hello @chetananakum31,
Thanks for reporting the issue, our team is looking into it. It would be great for us if you can share the below things to understand the cause properly.

*  Library version
*  Screenshot/video with steps to reproduce the issue
*  Provide the full logs

**Comment by chetananakum31** [comment]
@Nkgohil007
Sorry for delay .. As you requested, please find the logs and video.
There are several files but I trimmed the log as it has thousands of lines (contain scan file image name)

1) **Library version** -v 0.2.2 (Master)
2) **Video**

https://user-images.githubusercontent.com/93535061/148074614-0dd00da6-8da0-4778-82fe-9b5f0a2c28bb.mp4

3) **Logs**

```
V/PhoneWindow: DecorView setVisiblity: visibility = 0, Parent = android.view.ViewRootImpl@e1287cf, this = DecorView@c5f155c[MainActivity]
D/ViewRootImpl: enqueueInputEventMotionEvent { action=ACTION_DOWN, actionButton=0, id[0]=0, x[0]=218.75, y[0]=261.5, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, classification=NONE, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=122480576, downTime=122480576, deviceId=4, source=0x1002, displayId=0 }
D/ViewRootImpl[MainActivity]: processMotionEvent MotionEvent { action=ACTION_DOWN, actionButton=0, id[0]=0, x[0]=218.75, y[0]=261.5, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, classification=NONE, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=122480576, downTime=122480576, deviceId=4, source=0x1002, displayId=0 }
D/ViewRootImpl[MainActivity]: dispatchPointerEvent handled=true, event=MotionEvent { action=ACTION_DOWN, actionButton=0, id[0]=0, x[0]=218.75, y[0]=261.5, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, classification=NONE, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=122480576, downTime=122480576, deviceId=4, source=0x1002, displayId=0 }
D/ViewRootImpl[MainActivity]: processMotionEvent MotionEvent { action=ACTION_UP, actionButton=0, id[0]=0, x[0]=218.75, y[0]=261.5, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, classification=NONE, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=122480637, downTime=122480576, deviceId=4, source=0x1002, displayId=0 }
D/ViewRootImpl[MainActivity]: dispatchPointerEvent handled=true, event=MotionEvent { action=ACTION_UP, actionButton=0, id[0]=0, x[0]=218.75, y[0]=261.5, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, classification=NONE, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=122480637, downTime=122480576, deviceId=4, source=0x1002, displayId=0 }
D/libMEOW: applied 0 plugin for [com.lassi.app].
V/PhoneWindow: DecorView setVisiblity: visibility = 4, Parent = null, this = DecorView@a0554c8[]
D/ViewRootImpl[LassiMediaPickerActivity]: hardware acceleration = true , fakeHwAccelerated = false, sRendererDisabled = false, forceHwAccelerated = false, sSystemRendererDisabled = false
I/InputTransport: Create ARC handle: 0xb400007ba594ada0
V/PhoneWindow: DecorView setVisiblity: visibility = 0, Parent = android.view.ViewRootImpl@cb3f947, this = DecorView@a0554c8[LassiMediaPickerActivity]
V/OplusZoomWindowDecorViewHelper: setLastReportedMergedConfiguration mZoomDisplayHeight: 2400 getDecorView.DecorView@a0554c8[LassiMediaPickerActivity]
V/PhoneWindow: DecorView setVisiblity: visibility = 4, Parent = android.view.ViewRootImpl@e1287cf, this = DecorView@c5f155c[MainActivity]
E/MediaRepositoryImpl: IMG-20220104-WA0005 >> 138545
E/MediaRepositoryImpl: IMG-20220104-WA0004 >> 143015
E/MediaRepositoryImpl: IMG-20220104-WA0018 >> 151088
E/MediaRepositoryImpl: IMG-20220104-WA0017 >> 120422
E/MediaRepositoryImpl: IMG-20220104-WA0016 >> 95450
E/MediaRepositoryImpl: Screenshot_2022-01-03-15-49-28-15_b1f7bbf91b565db9420d418963bac8aa >> 612246
E/MediaRepositoryImpl: Screenshot_2022-01-03-14-24-42-78_a23b203fd3aafc6dcb84e438dda678b6 >> 763466
E/MediaRepositoryImpl: IMG20220102191839 >> 2919301
E/MediaRepositoryImpl: IMG20220102191004 >> 2985665
E/MediaRepositoryImpl: IMG_7159 >> 2110598
E/MediaRepositoryImpl: IMG_7153 >> 2620936
E/MediaRepositoryImpl: IMG_6634 >> 776916
E/MediaRepositoryImpl: IMG_6635 >> 1436808
E/MediaRepositoryImpl: IMG_7147 >> 2065823
E/MediaRepositoryImpl: IMG_7259 >> 1603031
E/MediaRepositoryImpl: GOPR4738 >> 2435968
E/MediaRepositoryImpl: GOPR4739 >> 2439479
E/MediaRepositoryImpl: GOPR4740 >> 2378272
E/MediaRepositoryImpl: GOPR4741 >> 2269496
E/MediaRepositoryImpl: GOPR4743 >> 2292662
E/MediaRepositoryImpl: GOPR4744 >> 2281033
E/MediaRepositoryImpl: DSC_0173 >> 6986514
E/MediaRepositoryImpl: DSC_0176 >> 7232613
    DSC_0177 >> 7017606
E/MediaRepositoryImpl: IMG20211211113222 >> 7384242
E/MediaRepositoryImpl: IMG20211211113208 >> 5327602
E/MediaRepositoryImpl: IMG20211211113202 >> 7282928
    IMG20211211113141 >> 6230579
E/MediaRepositoryImpl: IMG20211211113138 >> 6008072
E/MediaRepositoryImpl: IMG20211211113103 >> 6273446
    IMG20211211105514 >> 4475572
E/MediaRepositoryImpl: IMG20211211105506 >> 4165916
    IMG20211211084055 >> 3642880
E/MediaRepositoryImpl: IMG20211211084014 >> 5242842
    IMG20211211084010 >> 5748682
E/MediaRepositoryImpl: IMG20211211083959 >> 5262676
    IMG20211211083950 >> 5750435
E/MediaRepositoryImpl: IMG20211211083925 >> 3876111
    IMG20211211083848 >> 3542415
E/MediaRepositoryImpl: IMG20211211083839 >> 4664946
E/MediaRepositoryImpl: IMG20211211083744 >> 4076866
E/MediaRepositoryImpl: IMG20211211083740 >> 3987884
    IMG20211211083712 >> 3636713
E/MediaRepositoryImpl: IMG20211211083645 >> 3557186
    IMG20211211083636 >> 3155935
    IMG20211211083613 >> 3961942
E/MediaRepositoryImpl: DSC_0116 >> 4359502
E/MediaRepositoryImpl: DSC_0004 >> 4726811
E/MediaRepositoryImpl: DSC_0003 >> 4852507
E/MediaRepositoryImpl: DSC_0117 >> 4494261
E/MediaRepositoryImpl: DSC_0115 >> 4170632
E/MediaRepositoryImpl: DSC_0119 >> 4321228
E/MediaRepositoryImpl: DSC_0118 >> 4510841
E/MediaRepositoryImpl: DSC_0005 >> 4692510
E/MediaRepositoryImpl: DSC_0002 >> 4334200
E/MediaRepositoryImpl: IMG-20211211-WA0000 >> 120252
E/MediaRepositoryImpl: IMG-20211210-WA0006 >> 1952349
E/MediaRepositoryImpl: IMG-20211210-WA0004 >> 1962921
    DSC_0202 >> 4147470
    DSC_0201 >> 4314179
E/MediaRepositoryImpl: IMG-20190717-WA0010 >> 68389
E/MediaRepositoryImpl: IMG-20190720-WA0002 >> 54766
E/MediaRepositoryImpl: IMG-20190802-WA0006 >> 147351
E/MediaRepositoryImpl: IMG-20190802-WA0007 >> 100016
E/MediaRepositoryImpl: IMG-20190805-WA0001 >> 1359066
E/MediaRepositoryImpl: IMG-20190809-WA0009 >> 48173
I/com.lassi.app: Wrote stack traces to tombstoned
E/MediaRepositoryImpl: IMG-20190812-WA0004 >> 141140
E/MediaRepositoryImpl: IMG-20190812-WA0005 >> 111488
E/MediaRepositoryImpl: IMG-20190814-WA0007 >> 89623
E/MediaRepositoryImpl: IMG-20190814-WA0008 >> 106387
E/MediaRepositoryImpl: IMG-20190815-WA0011 >> 193033
Connected to process 23384 on device 'oneplus-dn2101-AYCUBY9TK7CAE6JB'.
Capturing and displaying logcat messages from application. This behavior can be disabled in the "Logcat output" section of the "Debugger" settings page.
I/oplus.android.OplusFrameworkFactoryImpl: get feature:IOplusScreenModeFeature
    getOplusScreenModeFeature
E/libc: Access denied finding property "ro.odm.prev.product.name"
D/NetworkSecurityConfig: No Network Security Config specified, using platform default
    No Network Security Config specified, using platform default
I/MultiDex: VM with version 2.1.0 has multidex support
    Installing application
    VM has multidex support, MultiDex support library is disabled.
D/libMEOW: meow reload base cfg path: na
    meow reload overlay cfg path: na
    applied 0 plugin for [com.lassi.app].
I/oplus.android.OplusFrameworkFactoryImpl: getOplusRedPacketManager
W/com.lassi.app: Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (greylist, reflection, allowed)
    Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (greylist, reflection, allowed)
E/OplusCustomizeRestrictionManager: sInstance is null, start a new sInstance
D/OplusInputMethodUtil: init DEBUG to false, update DEBUG_IME to false
I/SurfaceFactory: [static] sSurfaceFactory = com.mediatek.view.impl.SurfaceFactoryImpl@9ed2706
I/OplusFeatureCache: Milliseconds spent on init(): 7
D/OplusSlideAnimOptHelper: scrollEffectOpt enable = false
D/ViewRootImpl[MainActivity]: hardware acceleration = true , fakeHwAccelerated = false, sRendererDisabled = false, forceHwAccelerated = false, sSystemRendererDisabled = false
D/libMEOW: applied 0 plugin for [com.lassi.app].
I/InputTransport: Create ARC handle: 0xb400007b90f44180
V/PhoneWindow: DecorView setVisiblity: visibility = 0, Parent = android.view.ViewRootImpl@d351b60, this = DecorView@a341819[MainActivity]
V/OplusZoomWindowDecorViewHelper: setLastReportedMergedConfiguration mZoomDisplayHeight: 2400 getDecorView.DecorView@a341819[MainActivity]
D/libMEOW: applied 0 plugin for [com.lassi.app].
E/ion: ioctl c0044901 failed with code -1: Invalid argument
```

**Comment by sanjay-mi** [comment]
Hello @chetananakum31 

Thanks for using this library.

We have resolved the issue in latest lassi version 0.3.0

Thank you
